build.yaml

```diff
@@ -196,13 +215,25 @@ common_cli:
     # SSL on our caching box.
     # The --trusted-host option is only available with pip 7.
     #
-    parse_logs pip install --upgrade pip
-    # using the caching-layer, install all the dependencies
-    parse_logs pip install -i \$PIP_INDEX_URL . --trusted-host \$TRUSTED_HOST
-    # using the caching-layer, install flocker
-    parse_logs pip install -i \$PIP_INDEX_URL "Flocker[dev]" --trusted-host \$TRUSTED_HOST
-    # using the caching-layer, install junix for our coverage report
-    parse_logs pip install -i \$PIP_INDEX_URL python-subunit junitxml  --trusted-host \$TRUSTED_HOST
+    if [ "\$(os)" == "ubuntu" ] || [ "\$(os)" == "centos" ]; then
+        parse_logs pip install --upgrade pip
+        # using the caching-layer, install all the dependencies
+        parse_logs pip install -i \$PIP_INDEX_URL . --trusted-host \$TRUSTED_HOST
+        # using the caching-layer, install flocker
+        parse_logs pip install -i \$PIP_INDEX_URL "Flocker[dev]" --trusted-host \$TRUSTED_HOST
+        # using the caching-layer, install junix for our coverage report
+        parse_logs pip install -i \$PIP_INDEX_URL python-subunit junitxml  --trusted-host \$TRUSTED_HOST
+    fi
+    if [ "\$(os)" == "OSX" ]; then
+      # we can't use our pip caching layer on our OSX cloudbees boxes
+      # as we're unable run the slave_setup plugin that injects the environment
+      # variables to the slave.
+      # So we just go upstream for the python modules.
+      pip install --upgrade pip
```

Comment:
Why are these not `parse_logs` as above?

Ideally we'd have variables set for the `PIP_INDEX_URL` and `TRUSTED_HOST` for OS X (say, `https://pypi.python.org/simple` and another default). Then we could avoid repeating the commands here (can be an XXX and follow up later).